File: backend/app/schemas/news.py
from pydantic import BaseModel, Field


class NewsItem(BaseModel):
    id: str
    title: str
    url: str = ""
    source: str = ""
    published_at: str = ""
    collected_at: str = ""
    image_url: str | None = None
    is_model_related: bool = False
    model_relevance_reason: str = ""
    model_name: str = "Unknown"
    provider: str = "Unknown"
    model_type: str = ""
    summary: str = ""
    impact: str = ""
    action: str = ""
    importance: str = "B"
    tags: list[str] = Field(default_factory=list)
    language: str = "ja"
    image_generated_url: str | None = None
    image_generated_alt: str | None = None
    image_generated_prompt: str | None = None

    model_config = {"extra": "ignore"}

File: frontend/src/types/news.ts
export interface NewsItem {
  id: string
  title: string
  url: string
  source: string
  published_at: string
  collected_at: string
  image_url: string | null
  is_model_related: boolean
  model_relevance_reason: string
  model_name: string
  provider: string
  model_type: string
  summary: string
  impact: string
  action: string
  importance: 'A' | 'B'
  tags: string[]
  language: string
  image_generated_url: string | null
  image_generated_alt: string | null
  image_generated_prompt: string | null
}
